src/odemis/acq/stitching/test/weaver_test.py

```diff
@@ -72,13 +140,78 @@ def test_one_tile(self):
         }
         intile = model.DataArray(img8, md8)
 
-        outd = collageWeaver([intile])
+        outd = weave([intile])
 
         self.assertEqual(outd.shape, intile.shape)
         numpy.testing.assert_array_equal(outd, intile)
         self.assertEqual(outd.metadata, intile.metadata)
+        
+    def test1(self):
+        """
+        Test on synthetic image
+        """
+
+        img = Image.open("images/test3.tiff")
+        numTiles = 2
+        overlap = 0.2
+        [tiles,pos] = decomposeImageExact(img,overlap,numTiles)
+        
+        weaver = MeanWeaver()
+        for i in range(len(pos)):
+            weaver.addTile(tiles[i])
+        
+        sz = len(weaver.getFullImage())
+        self.assertTrue((numpy.array_equal(weaver.getFullImage(),numpy.array(img)[:sz,:sz])))
+        
```

Comment:
Add at least one unit test where each tile is just a constant (128, 256, 384..), and check afterwards that the seams are gradients.